operator/controllers/loki/lokistack_controller.go

```diff
@@ -178,7 +187,8 @@ func (r *LokiStackReconciler) buildController(bld k8s.Builder) error {
 		Owns(&rbacv1.ClusterRole{}, updateOrDeleteOnlyPred).
 		Owns(&rbacv1.ClusterRoleBinding{}, updateOrDeleteOnlyPred).
 		Owns(&rbacv1.Role{}, updateOrDeleteOnlyPred).
-		Owns(&rbacv1.RoleBinding{}, updateOrDeleteOnlyPred)
+		Owns(&rbacv1.RoleBinding{}, updateOrDeleteOnlyPred).
+		Watches(&source.Kind{Type: &corev1.Service{}}, r.enqueueAllLokiStacksHandler(), createUpdateOrDeletePred)
```

Comment:
:scream: Why is this possible at all? Using the same source Kind for owned and watched resources?

TL;DR; Although the controller runtime API seems to allow this I suggest to dive into the code behind this to see what happens with caching, listers and informers when you register the same source Kind twice. This seems odd and dangerous.

Besides this the `Watches` clause seems to be also very eager sort of that any service across the cluster will trigger a reconciliation loop for all LokiStack instances. Can't we limit this to the two particular OCP services and only for the LokiStacks with tenant modes `openshift-logging` and `openshift-network`?

Reply:
I agree that it's reckless to have all the SVCs trigger the reconciliation when we are interested in only one specific SVC, so I'll edit this to be more specific. 

However, I don't see an issue with having an Owns and a Watches on the same resource; Owns is just a Watches under the hood, and if I'm not mistaken, we can have multiple Watches on the same resource in order to react to different events differently. If it's not possible then please enlighten me as I might have misunderstood the behavior.